SOLR-16824: Adopt Linux Command line tool pattern of -- for long option commands (#1768)
apache/solr: solr/solr-ref-guide/modules/upgrade-notes/pages/major-changes-in-solr-10.adoc

// TODO add similar text that previous releases have at this spot.

=== Solr CLI and Scripts

> The Solr CLI has gone through some signficant rennovations to reduce technical debt, and function more consistently and predictably.

Reviewer: ```suggestion
The Solr CLI has gone through some significant renovations to reduce technical debt, and now functions more consistently and predictably. Most notably, CLI commands now use unix-style options, e.g. `--help` instead of `-help`.
```

> You will need to review any custom scripts that use the `bin/solr` command to ensure that you are using the correct options. 

Reviewer: ```suggestion
Users are urged to review all use of the `bin/solr` command, such as in scripts and documentation, to ensure that the correct options are being used.
```

* The `<hiddenSysProps>` solr.xml element under `<metrics>` has been removed. Instead use the `<hiddenSysProps>` tag under `<solr>`, which accepts a comma-separated string.
Please see `-Dsolr.redaction.system.pattern`, which allows users to provide a pattern to match sysProps that should be redacted for sensitive information,
has been removed. Please use `-Dsolr.hiddenSysProps` or the envVar `SOLR_HIDDEN_SYS_PROPS` instead.

=== SolrCLI

The Solr CLI has gone through some signficant rennovations to reduce technical debt, and function more consistently and predictably.
You will need to review any custom scripts that use the `bin/solr` command to ensure that you are using the correct options. 

> Here are changes from Solr 9.3 and Solr 10 in the CLI.

Reviewer: You can remove this line, as these changes are only from 10.0 onwards?

Author: Correct...   This is way to hard to make be backward compatible...     Do I need to list out the before or after in the changelog, or would it be enough to say "See the Reference Guide page XXX for the long parameter names for each command."

Author: I've made the change that @janhoy  suggested!
